src/devices/bus/epson_qx/option.h

```diff
@@ -0,0 +1,276 @@
+    |   |  |   |  |   |  |   |  |   |
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------IRQH2 (Master IR3)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  |   |  | *-|--------IRQL  (Slave  IR1)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  | *-|--|---|--------IRQL  (Slave  IR3)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  | *-|--|---|--|---|--------IRQL  (Slave  IR4)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  | *-|--|---|--|---|--|---|--------IRQL  (Slave  IR6)
+    |   |  |   |  |   |  |   |  |   |
+    | *-|--|---|--|---|--|---|--|---|--------IRQL  (Slave  IR7)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  |   |  |   |
+    |---|  |---|  |---|  |---|  |---|
+
+    The DREQ/DACK(F) signals are the high priorty DMA signals and are shared between all option slots.
+    The DREQ/DACK(S) signals are the lower priority DMA signals and are local to each slot, with the
+    exception of slot 5 which does not provide low priority DMA signals. The RDY/EOP(S) signals are
+    also connected to all 5 options slots since the DREQ(S) signals are all connected to the same
+    DMA controller.
+
+      5      4      3      2      1
+    |---|  |---|  |---|  |---|  |---|
+    |   |  |   |  |   |  |   |  |   |
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------DREQF (Master CH3)
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------DACKF (Master Ch3)
+    |   |  |   |  |   |  |   |  |   |
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------RDYF  (Master RDY)
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------EOPF  (Master EOP)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  |   |  | *-|--------DREQS (Slave  CH0)
+    |   |  |   |  |   |  |   |  | *-|--------DACKS (Slave  CH0)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  |   |  | *-|--|---|--------DREQS (Slave  CH1)
+    |   |  |   |  |   |  | *-|--|---|--------DACKS (Slave  CH1)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  |   |  | *-|--|---|--|---|--------DREQS (Slave  CH2)
+    |   |  |   |  | *-|--|---|--|---|--------DACKS (Slave  CH2)
+    |   |  |   |  |   |  |   |  |   |
+    |   |  | *-|--|---|--|---|--|---|--------DREQS (Slave  CH3)
+    |   |  | *-|--|---|--|---|--|---|--------DACKS (Slave  CH3)
+    |   |  |   |  |   |  |   |  |   |
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------RDYS  (Slave RDY)
+    | *-|--|-*-|--|-*-|--|-*-|--|-*-|--------EOPS  (Slave EOP)
+    |   |  |   |  |   |  |   |  |   |
+    |---|  |---|  |---|  |---|  |---|
+
+*/
+#ifndef MAME_BUS_EPSON_QX_BUS_H
+#define MAME_BUS_EPSON_QX_BUS_H
+
+#pragma once
+
+#include <vector>
+
+//**************************************************************************
+//  FORWARD DECLARATIONS
+//**************************************************************************
+
+class option_bus_device;
+class device_option_expansion_interface;
```

Comment:
If you’re going to use short, generic names like this, you’ll need to put the contents of this folder in a namespace to avoid conflicts.  I’d suggest `namespace bus::epson_qx`.